backend/src/main/java/de/metanome/backend/algorithm_execution/AlgorithmExecutor.java

```diff
@@ -130,11 +163,15 @@ public long executeAlgorithmWithValues(de.metanome.backend.results_db.Algorithm
     if (analyzer.isFunctionalDependencyAlgorithm()) {
       FunctionalDependencyAlgorithm fdAlgorithm = (FunctionalDependencyAlgorithm) algorithm;
       fdAlgorithm.setResultReceiver(resultReceiver);
+
```

Comment:
algorithm should have type-Enum and analyzer EnumMap<algorithm.type, boolean> - iterate over possible types and if true do something like:
algorithm.setResultReceiver(resultReceiver);
results.add(new Result(resultPathPrefix + ResultPrinter.getEnding(type)).setType(type);
- should abvoid a lot of code duplication!